src/solve.jl

```diff
@@ -119,6 +119,31 @@ function solve(system::InitialValueProblem{<:HybridSystem, <:LazySet{N}},
     return solve!(sys_new, copy(options), opC, opD)
 end
 
+
+"""
+get_necessary_vars(HS::HybridSystem):: Vector{Int64}
+
+Function to get only variables which are used for guard and invariant constraints
+
+### Input
+
+- `HS`  -- hybrid system
+"""
+function get_necessary_vars(HS::HybridSystem):: Vector{Int64}
+   vars = Vector{Int64}()
```

Comment:
note that if there are many constraints, several new, temporary `vars` arrays have to be allocated in the inner loops.

 perhaps you can consider working with the same array all the way and use `unique` only once, in the end. that is, to substitute calls to `vcat` and assignments with calls `push!(vars, ...)` and one call to `unique` to filter repeated values.